### PR TITLE
19847 - Split UsersSection into components for better maintainability

### LINKED ISSUE #19847: Split UsersSection into components for better maintainability
### Description

As discussed in #17728

### PR DESCRIPTION
**Pull request by andreas-blaettlinger**
Closes #19847

As the UsersSection component is getting quite large, a split into sub-components was necessary to improve maintainability.

### CODE REVIEW

File: js/apps/admin-ui/src/components/users/UserDataTableToolbarItems.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import { useAccess } from "../../context/access/Access";
+
+import type RealmRepresentation from "@keycloak/keycloak-admin-client/lib/defs/realmRepresentation";
+
+type UserDataTableToolbarItemsProps = {
+  realm: RealmRepresentation;
+  hasSelectedRows: boolean;
+  toggleDeleteDialog: any;
+  toggleUnlockUsersDialog: any;
+  goToCreate: any;
```
Comment (edewit) [inline js/apps/admin-ui/src/components/users/UserDataTableToolbarItems.tsx:20]: We don't want to use `any` as this defeats the purpose of using typescript

File: js/apps/admin-ui/src/components/users/UserDataTableToolbarItems.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+  const { hasAccess } = useAccess();
+
+  // Only needs query-users access to attempt add/delete of users.
+  // This is because the user could have fine-grained access to users
+  // of a group.  There is no way to know this without searching the
+  // permissions of every group.
+  const isManager = hasAccess("query-users");
+
+  const bruteForceProtecionToolbarItem = !realm.bruteForceProtected ? (
```
Comment (edewit) [inline js/apps/admin-ui/src/components/users/UserDataTableToolbarItems.tsx:41]: spelling error
```suggestion
  const bruteForceProtectionToolbarItem = !realm.bruteForceProtected ? (
```

**Review by edewit — CHANGES_REQUESTED** [review]
great contribution fix up these small issues and then we can merge this

File: js/apps/admin-ui/src/components/users/UserDataTableToolbarItems.tsx (hunk 1 above)
Comment (andreas-blaettlinger) [reply js/apps/admin-ui/src/components/users/UserDataTableToolbarItems.tsx]: Fixed

File: js/apps/admin-ui/src/components/users/UserDataTableToolbarItems.tsx (hunk 2 above)
Comment (andreas-blaettlinger) [reply js/apps/admin-ui/src/components/users/UserDataTableToolbarItems.tsx]: Fixed

**Comment by andreas-blaettlinger** [comment]
@edewit I updated PR with suggested fixes. Any further feedback?

**Review by edewit — APPROVED** [review]

**Comment by andreas-blaettlinger** [comment]
@edewit There are some test errors but they are not related to my changes. Can you merge it anyway?

**Review by jonkoops — APPROVED** [review]

**Comment by jonkoops** [comment]
Thanks for this @andreas-blaettlinger! And apologies for the late review, things have been a bit bogged down recently.

**Pull request merged**